dotnet/src/SemanticKernel.Abstractions/Events/FunctionInvokingEventArgs.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Events;
+
+/// <summary>
+/// Event arguments available to the Kernel.FunctionInvoking event.
+/// </summary>
+public class FunctionInvokingEventArgs : SKCancelEventArgs
+{
+    /// <summary>
+    /// Indicates if the function execution should be skipped.
+    /// </summary>
+    public bool IsSkipRequested => this._skipRequested;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FunctionInvokingEventArgs"/> class.
+    /// </summary>
+    /// <param name="functionView">Function view details</param>
+    /// <param name="context">Context related to the event</param>
+    public FunctionInvokingEventArgs(FunctionView functionView, SKContext context) : base(functionView, context)
+    {
+    }
+
+    /// <summary>
+    /// Skip the current function invoking attempt.
+    /// </summary>
+    public void Skip()
```

Comment:
thought: semantic functions call other functions, and we do the rendering (and therefore calling of other functions) before we give them the chance to skip.

Reply:
Yep, you are right, currently event's don't trigger when Rendering (Even if the template makes reference to other Semantic/Native functions.

There are ongoing discussions how to solve that:

1. Add more rendering related events to Kernel
2. Add rendering events to ITemplateEngine interface
3. Add the rendering callbacks to the current Invoked/Invoking